### PR TITLE
Adds FluentIconSwitch component, ensures correct environment variable masking behavior when the resource panel is open

### LINKED ISSUE #1519: Dashboard's details pane can become inconsistent when viewing environment for different resources
In the Dashboard, when viewing the environment in the Resources view, the state of the **Show Values** toolbar button in the Details pane is inconsistent when switching between resources. Taking the starter application as an example:

- In the Resources view, click **View** in the Environment column for the the `apiservice` resource. 
- Click **Show Values** in the Details view toolbar. Notice that the button changes to **Hide Values** and that all the environment variable values are now shown. 
 
  image

- Now click **View** in the Environment column for the the `webfrontend` resource. Notice that the toolbar button still appears as **Hide Values** but that the environment variable values remain hidden. It's then necessary to click the button again twice to show the values.

  image

Ideally, the state of this setting should apply to all resources, at least until you switch to a different a different page. Otherwise, if the intention is to always start from a hidden state, then the button needs to be updated when switching resources.

### PR DESCRIPTION
**Pull request by adamint**
We have two buttons being used as switches - fluentui-blazor's switch component does not allow for custom icons, so I created `FluentIconSwitch` so that this behavior is encompassed in a component for reuse.

Fixes #1519 by updating individual environment variable `IsValueMasked` on component render
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1700)

### CODE REVIEW

File: src/Aspire.Dashboard/Components/Controls/FluentIconSwitch.razor [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿<FluentButton Appearance="@Appearance"
+              Disabled="Disabled"
+              Title="@(Value is true ? CheckedTitle : UncheckedTitle)"
+              aria-label="@(Value is true ? CheckedTitle : UncheckedTitle)"
+              OnClick="@(() => OnToggleInternalAsync())"
+              IconEnd="@(Value is true ? CheckedIcon : UncheckedIcon)"
+              slot="end" />
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/FluentIconSwitch.razor:7]: `slot` shouldn't be part of the component. It should be an attribute you can add to the component and is passed to the fluent button.

Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Controls/FluentIconSwitch.razor:7]: Something like this:

* https://github.com/microsoft/fluentui-blazor/blob/ef53ebde9544fd1b58b8ceca6744d1860f459285/src/Core/Components/Base/FluentComponentBase.cs#L54-L58
* https://github.com/microsoft/fluentui-blazor/blob/ef53ebde9544fd1b58b8ceca6744d1860f459285/src/Core/Components/Button/FluentButton.razor#L47

File: src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs [hunk 2]
```diff
@@ -174,12 +180,11 @@ private static string GetDisplayedValue(SummaryValue summaryValue)
 
     private void ToggleMaskState()
     {
-        _defaultMasked = !_defaultMasked;
-        if (Resource.Environment is { } environment)
+        if (Resource.Environment is var environment)
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/Controls/ResourceDetails.razor.cs:183]: I haven't seen this before. Does it combine a not null check?

**Review by kvenkatrajan — APPROVED** [review]

File: src/Aspire.Dashboard/Components/Controls/FluentIconSwitch.razor (hunk 1 above)
Comment (adamint) [reply src/Aspire.Dashboard/Components/Controls/FluentIconSwitch.razor]: This has been addressed

**Pull request merged**